Tests/Test_CriticalState_1.c

```diff
@@ -0,0 +1,135 @@
+/* Copyright (c) 2021 UT Longhorn Racing Solar */
+
+#include "common.h"
+#include "config.h"
+#include "os.h"
+#include "Tasks.h"
+#include "stm32f4xx.h"
+#include "BSP_Contactor.h"
+#include "BSP_Lights.h"
+#include "BSP_PLL.h"
+
+OS_TCB Task1_TCB;
+CPU_STK Task1_Stk[256];
+OS_TCB Task2_TCB;
+CPU_STK Task2_Stk[256];
+
+/******************************************************************************
+ * Critical State Task Test Plan
+ * 
+ * 1. Build the BPS code for the stm32f413 for this test file (see README.md)
+ * 2. Flash the BPS (see README.md)
+ * 3. Connect the BPS UART USB to your laptop and open PuTTY (or comparable)
+ * 4. Connect the BPS test board
+ * 5. Run the code in this file
+ * 6. Make sure the contactor stays closed
+****************************************************************************/
+
+void Task2(void *p_arg){    //This task is meant to allow contactor to close
+    OS_ERR err;
+
+    OSTimeDly(250, OS_OPT_TIME_DLY, &err);
+
+
+    OSSemPost(&SafetyCheck_Sem4,
+                OS_OPT_POST_1,
+                &err);
+    assertOSError(err);
+    
+    OSSemPost(&SafetyCheck_Sem4,
+                OS_OPT_POST_1,
+                &err);
+    assertOSError(err);
+
+    OSSemPost(&SafetyCheck_Sem4,
+                OS_OPT_POST_1,
+                &err);
+    assertOSError(err);
+
+    OSSemPost(&SafetyCheck_Sem4,
+                OS_OPT_POST_1,
+                &err);
+    assertOSError(err);
+
+    BSP_Lights_Init();
+   
+    while(1) {
+        
+        OSTimeDly(25, OS_OPT_TIME_DLY, &err);
+    }
```

Comment:
Why are you doing this?

Reply:
I forgot to add the light toggle statements, this is fixed in latest commit